narwhals/_spark_like/utils.py

```diff
@@ -112,9 +113,16 @@ def narwhals_to_native_dtype(
 
 def parse_exprs_and_named_exprs(
     df: SparkLikeLazyFrame,
-) -> Callable[..., dict[str, Column]]:
-    def func(*exprs: SparkLikeExpr, **named_exprs: SparkLikeExpr) -> dict[str, Column]:
+) -> Callable[..., tuple[dict[str, Column], list[bool]]]:
+    def func(
+        *exprs: SparkLikeExpr, **named_exprs: SparkLikeExpr
+    ) -> tuple[dict[str, Column], list[bool]]:
         native_results: dict[str, list[Column]] = {}
+
+        # `returns_scalar` keeps track if an expression returns a scalar and is not lit.
+        # Notice that lit is quite special case, since it gets broadcasted by pyspark
```

Comment:
Out of interest, do we run into any issues if we use `over` anyway with `lit`? No objections to special casing it, just curious

Reply:
We end up with `tests/expr_and_series/lit_test.py` failing 3 tests due to:

> pyspark.errors.exceptions.captured.AnalysisException: [UNSUPPORTED_EXPR_FOR_WINDOW] Expression "1" not supported within a window function.;

Reply:
isn't this still going to break for, say
```
df.with_columns(nw.lit(2)+1)
```
?

Reply:
Yes correct